backend/server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/ServerInfoSynchronizer.java

```diff
@@ -41,6 +45,16 @@ public void synchronize(ServerApi serverApi, SonarLintCancelMonitor cancelMonito
     var serverStatus = serverApi.system().getStatus(cancelMonitor);
     var serverVersionAndStatusChecker = new ServerVersionAndStatusChecker(serverApi);
     serverVersionAndStatusChecker.checkVersionAndStatus(cancelMonitor);
-    storage.serverInfo().store(serverStatus);
+    var isMQRMode = serverApi.isSonarCloud() || isMQRMode(serverApi.settings(), serverStatus.getVersion(), cancelMonitor);
```

Comment:
Niptick: I would argue that the first part of the OR condition should be in the `isMQRMode`. Or rename the 2nd part to `isSqMQRMode`